### PR TITLE
Move content validation to StorageTX, add TX principal name

### PR DESCRIPTION
**Pull request by cstamas**
More improvements to TX, such as:
- move-in content validation (and remove lot of dupe code from plugins)
- pre-fill constant Blob headers (instead doing it in plugins)

Builds upon
https://github.com/sonatype/nexus-oss/pull/1192

CI
http://bamboo.s/browse/NX3-OSSF328

### CODE REVIEW

File: components/nexus-repository/src/main/java/org/sonatype/nexus/repository/storage/AssetBlob.java [hunk 1]
```diff
@@ -61,6 +65,11 @@ void setAttached(final boolean attached) {
   }
 
   @Nonnull
+  public String getBlobName() {
```
Comment (mrprescott) [inline components/nexus-repository/src/main/java/org/sonatype/nexus/repository/storage/AssetBlob.java]: Is this used?

**Comment by mrprescott** [comment]
It makes sense to consolidate how content validation gets done. Is there any risk of formats where some content needs validating and some doesn't?

**Comment by mrprescott** [comment]
+1

**Comment by cstamas** [comment]
@mrprescott I too think this is good direction, but will close out this PR and create new one that will factor in format specific validators (based on similar solution as with HttpResponseSender).

**Pull request closed without merging**